src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -94,6 +95,53 @@ public Mono<Tuple2<Mono<TransactionActivatedEvent>, String>> handle(
+                                    return Mono.just(
+                                            Tuples.of(
+                                                    paymentNotice,
+                                                    Optional.of(
+                                                            maybePaymentRequestInfo
+                                                                    .filter(
+                                                                            requestInfo -> isValidIdempotencyKey(
+                                                                                    requestInfo.idempotencyKey()
+                                                                                            .rawValue()
+                                                                            )
+                                                                    )
+                                                                    .orElseGet(
+                                                                            () -> {
+                                                                                PaymentRequestInfo paymentRequestWithOnlyIdempotencyKey = new PaymentRequestInfo(
+                                                                                        new RptId(
+                                                                                                paymentNotice.getRptId()
+                                                                                        ),
+                                                                                        null,
+                                                                                        null,
+                                                                                        null,
+                                                                                        null,
+                                                                                        null,
+                                                                                        true,
+                                                                                        null,
+                                                                                        new IdempotencyKey(
+                                                                                                nodoOperations
+                                                                                                        .getEcommerceFiscalCode(),
+                                                                                                nodoOperations
+                                                                                                        .generateRandomStringToIdempotencyKey()
+                                                                                        )
+                                                                                );
```

Comment:
If we decide that these fields can be `null` I would add a `@Nullable` annotation on them inside `PaymentRequestInfo` (and maybe a constructor with non-null fields only?)

Reply:
correct! for a payment request we can also save only `<rptId,idempotencyKey>`, in order to handle retry with the same idempotencyKey. So, I have to to update `ecommerce-commons` -> https://github.com/pagopa/pagopa-ecommerce-commons/blob/main/src/main/java/it/pagopa/ecommerce/commons/repositories/PaymentRequestInfo.java

Reply:
Solved with pagopa/pagopa-ecommerce-commons#37